src/Game.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+
+function Game({ quest, onClickAnswer }) {
+  const { incorrect_answers, correct_answer, question } = quest;
+
+  const [selectedAnswer, setSelectedAnswer] = useState(null);
+
+  useEffect(() => {
+    setSelectedAnswer(null);
+  }, [quest]);
+
+  // const checkAnswer = (e, answer) => {
+  //   onClickAnswer(answer);
+  //   if (locked === false) {
+  //     if (answer === correct_answer) {
+  //       e.target.classList.add('correct');
+  //       setLocked(true);
+  //     } else {
+  //       e.target.classList.add('incorrect');
+  //       setLocked(true);
+  //       setCorrectAnswer(correct_answer);
+  //     }
+  //   }
+  // };
+
+  const handleAnswerClick = (answer) => {
+    if (!selectedAnswer) {
+      onClickAnswer(answer);
+      setSelectedAnswer(answer);
+    }
+  };
+
+  // const getAnswerClassName = (answer) => {
+  //   if (selectedAnswer) {
+  //     if (answer === correct_answer) {
+  //       return 'correct';
+  //     } else if (answer === selectedAnswer) {
+  //       return 'incorrect';
+  //     }
+  //   }
+  //   return '';
+  // };
+
+  const answers = [correct_answer, ...incorrect_answers].sort();
+
+  return (
+    <>
+      <h1>{quest.question}</h1>
+      <ul>
+        {answers.map((answer) => (
+          <li
+            onClick={() => handleAnswerClick(answer)}
+            style={{
```

Comment:
Look at the comments to the previous PR (the feedback one) to improve the list.

Reply:
Can you explain what should i change, the function or the style property?

Reply:
Move the logic away from the jsx as much as possible. Also, whenever you have 2 routes and you're not sure which one is best, start by doing something and showing your work, then we can discuss if it was the best option.

Here's my copy/pasted reply to the previous PR.

Instead of directly manipulating classes using classList, consider using React state to manage class names. This can be done by maintaining a state to track whether the answer is correct or not.

```jsx
const [selectedAnswer, setSelectedAnswer] = useState(null);

const checkAnswer = (answer) => {
  setSelectedAnswer(answer);
  onClickAnswer(answer);
};

// ...

<li
  onClick={() => checkAnswer(answer)}
  className={selectedAnswer === answer ? 'selected' : ''}
  key={answer}
>
  {answer}
</li>
```